Case Study on ANOVA

XYZ Company has offices in four different zones. The company wishes to
investigate the following :

● The mean sales generated by each zone.

● Total sales generated by all the zones for each month.

● Check whether all the zones generate the same amount of sales.

Help the company to carry out their study with the help of data provided.

In [62]:

#importing necessary modules
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
from google.colab import drive

Reading and preparing data

In [63]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [64]:
df= pd.read_csv('/content/drive/My Drive/Sales_data_zone_wise.csv')
df.head()     #reading dataset in py environment

Month  Zone - A  Zone - B  Zone - C  Zone - D
0  Month - 1   1483525   1748451   1523308   2267260
1  Month - 2   1238428   1707421   2212113   1994341
2  Month - 3   1860771   2091194   1282374   1241600
3  Month - 4   1871571   1759617   2290580   2252681
4  Month - 5   1244922   1606010   1818334   1326062

In [65]:
df.columns #displaying features

Index(['Month', 'Zone - A', 'Zone - B', 'Zone - C', 'Zone - D'], dtype='object')

In [66]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 29 entries, 0 to 28
Data columns (total 5 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   Month     29 non-null     object
 1   Zone - A  29 non-null     int64 
 2   Zone - B  29 non-null     int64 
 3   Zone - C  29 non-null     int64 
 4   Zone - D  29 non-null     int64 
dtypes: int64(4), object(1)
memory usage: 1.3+ KB


In [67]:
df.isna().sum()  #checking null values

Month       0
Zone - A    0
Zone - B    0
Zone - C    0
Zone - D    0
dtype: int64

In [68]:
df.describe()  # extracting statistical data

Zone - A      Zone - B      Zone - C      Zone - D
count  2.900000e+01  2.900000e+01  2.900000e+01  2.900000e+01
mean   1.540493e+06  1.755560e+06  1.772871e+06  1.842927e+06
std    2.619401e+05  1.683899e+05  3.331937e+05  3.750165e+05
min    1.128185e+06  1.527574e+06  1.237722e+06  1.234311e+06
25%    1.305972e+06  1.606010e+06  1.523308e+06  1.520406e+06
50%    1.534390e+06  1.740365e+06  1.767047e+06  1.854412e+06
75%    1.820196e+06  1.875658e+06  2.098463e+06  2.180416e+06
max    2.004480e+06  2.091194e+06  2.290580e+06  2.364132e+06

Findings:
• The given dataset is the zone wise monthly sales of a company. 

• There are 4 zones, A, B, C & D. and data of 29 months are given. 

•There are no null values in the data.
 
•There are 29 observations and a total of five columns with one object datatype and 4 int 64 datatype.

1) The Mean sales generated by each zone.

In [69]:
Mean_sales=df.mean(axis = 0) #storing in variable  Mean_sales
Mean_sales.round()

Zone - A    1540493.0
Zone - B    1755560.0
Zone - C    1772871.0
Zone - D    1842927.0
dtype: float64

Findings:

Mean/average sales generated by each zone is found out. 

•  Zone-D has more average sales compared to other zones. 

• Zone - B and Zone - C have almost same average sales. 

• Zone-A has least mean sales.

2) Total sales generated by all the zones for each month.

In [70]:
df['Total Sales'] =df.sum(axis = 1)
df

Month  Zone - A  Zone - B  Zone - C  Zone - D  Total Sales
0    Month - 1   1483525   1748451   1523308   2267260      7022544
1    Month - 2   1238428   1707421   2212113   1994341      7152303
2    Month - 3   1860771   2091194   1282374   1241600      6475939
3    Month - 4   1871571   1759617   2290580   2252681      8174449
4    Month - 5   1244922   1606010   1818334   1326062      5995328
5    Month - 6   1534390   1573128   1751825   2292044      7151387
6    Month - 7   1820196   1992031   1786826   1688055      7287108
7    Month - 8   1625696   1665534   2161754   2363315      7816299
8    Month - 9   1652644   1873402   1755290   1422059      6703395
9   Month - 10   1852450   1913059   1754314   1608387      7128210
10  Month - 11   2004480   1740365   1317869   1970069      7032783
11  Month - 12   1537539   1875658   1237722   1460165      6111084
12  Month - 13   1254939   1588473   1348629   1733383      5925424
13  Month - 14   1821799   1844081   1617376   1872259      7155515
14  Month - 15   1128185   1804613   1767047   1234311      5934156
15  Month - 16   1491534   1542053   1882159   1590913      6506659
16  Month - 17   1182110   1552388   2234469   2180416      7149383
17  Month - 18   1770324   1699213   1249821   2364132      7083490
18  Month - 19   1181635   1959268   2009168   1821882      6971953
19  Month - 20   1305972   1559670   2107949   2151008      7124599
20  Month - 21   1570152   2013615   1540016   2265814      7389597
21  Month - 22   1481619   1527574   2255729   2295079      7560001
22  Month - 23   1887694   1992155   1348387   1459683      6687919
23  Month - 24   1880820   1752873   2098463   2052591      7784747
24  Month - 25   1256333   1622671   1521792   1695122      6095918
25  Month - 26   1422853   1715465   1853636   1520406      6512360
26  Month - 27   1384426   1983163   1611169   1289160      6267918
27  Month - 28   1616640   1547991   2128022   2178267      7470920
28  Month - 29   1310654   1660092   1947119   1854412      6772277

In [73]:

#displaying minimum value of sales. 
df['Total Sales'].min()

5925424

In [74]:
df['Total Sales'].max()

8174449

Findings:

•Maximum sales is generated on 4th month(8174449) 

Minimum sales generated on 13th month(5925424).

3) Check whether all the zones generate the same amount of sales.


Here we have to check one-way Anova test. 

Stating hypothesis;

Null hypothesis H0: Amount of sales from all zones are equal. 

Alternate hypothesis Ha:Amount of sales from all zones are not equal. 

Let Significance level: 0.05

In [75]:
import scipy
from scipy import stats
from scipy.stats import f_oneway

In [76]:
F,p=scipy.stats.f_oneway(df['Zone - A'],df['Zone - B'], df['Zone - C'], df['Zone - D'])
print(['F-statistic:',F]) 
print(['p-value:',p])
if p > 0.05:
    print('Accept Null Hypothesis')
else :
    print ('Reject Null Hypothesis')

['F-statistic:', 5.672056106843581]
['p-value:', 0.0011827601694503335]
Reject Null Hypothesis


Findings:

• F- statistic = 5.6720 and  
p-value =0.00118<0.05

Hence we can reject null hypothesis and accept alternate hypothesis which states that all zones have different sales amount.